cpp/ql/test/library-tests/ir/ir/ir.cpp

```diff
@@ -1006,4 +1006,25 @@ void OperatorDeleteArray() {
 }
 #endif
 
+struct EmptyStruct {};
+
+void EmptyStructInit() {
+  EmptyStruct s = {};
+}
+
+auto lam = []() {};
+
+void Lambda(int x, const String& s) {
+  auto lambda_empty = [](float f) { return 'A'; };
+  lambda_empty(0);
+  auto lambda_ref = [&](float f) { return s.c_str()[x]; };
+  lambda_ref(1);
+  auto lambda_val = [=](float f) { return s.c_str()[x]; };
+  lambda_val(2);
+  auto lambda_ref_explicit = [&s](float f) { return s.c_str()[0]; };
+  lambda_ref_explicit(3);
+  auto lambda_val_explicit = [s](float f) { return s.c_str()[0]; };
+  lambda_val_explicit(4);
```

Comment:
Can you add a test with multiple captured variables, ideally with different capture modes. Is the translation intended to support [the full range of C++17 capture modes](https://en.cppreference.com/w/cpp/language/lambda#Lambda_capture)?

Reply:
Done. Captures with initializers, which I did not know existed until now, work with no changes. Explicit captures of `this` produce some incorrect-looking ASTs, so I'll have to dig into that separately. I've added an `#ifd'd-out test case for those, and will investigate further after this PR is merged.